libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -417,6 +417,40 @@ void AP_DDS_Client::update_topic(geographic_msgs_msg_GeoPoseStamped& msg)
     }
 }
 
+void AP_DDS_Client::update_topic(sensor_msgs_msg_Imu& msg)
+{
+    update_topic(msg.header.stamp);
+    strcpy(msg.header.frame_id, BASE_LINK_FRAME_ID);
+
+    auto &imu = AP::ins();
+    auto &ahrs = AP::ahrs();
+
+    WITH_SEMAPHORE(ahrs.get_semaphore());
+    
+    Quaternion orientation;
+    if (ahrs.get_quaternion(orientation)) {
+        msg.orientation.x = orientation[0];
+        msg.orientation.y = orientation[1];
+        msg.orientation.z = orientation[2];
+        msg.orientation.w = orientation[3];
+    }
+    msg.orientation_covariance[0] = -1;
+
+    Vector3f accel_data = imu.get_accel();
```

Comment:
Do you only output the value of one IMU, or are there multiple to publish?

Reply:
I believe ins library returns data avg from all the imus. I'll have to confirm this though. However, it's sure that this data is raw. Ahrs data (used for orientation) is however filtered

Reply:
Looks good now.